proposals/0307-GetInputScreen.md

```diff
@@ -102,40 +128,175 @@ This enables text input fields, numeric input fields and combo box input fields.
+</param>
+
+<param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
+    <description>Provides additional human readable info regarding the result.</description>
+</param>
+
+<param name="resultCode" type="Result" platform="documentation" mandatory="true">
+    <description>See Result</description>
+    <element name="SUCCESS"/>
+    <element name="REJECTED"/>
+    <element name="INVALID_DATA"/>
+    <element name="INVALID_ID"/>
+    <element name="DUPLICATE_NAME"/>
+    <element name="DISALLOWED"/>
+    <element name="OUT_OF_MEMORY"/>
+    <element name="TOO_MANY_PENDING_REQUESTS"/>
+    <element name="APPLICATION_NOT_REGISTERED"/>
+    <element name="GENERIC_ERROR"/>
+</param>
 </function>
 ```
 
-We can use the `maskInputCharacters` parameter from [0238-Keyboard-Enhancements](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0238-Keyboard-Enhancements.md) for password or other sensitive data input.
+We can use the `maskInputCharacters` parameter from [0238-Keyboard-Enhancements](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0238-Keyboard-Enhancements.md) for password or other sensitive data input. The main goal of this is to leverage an existing enum to help us out. `maskInputCharacters` mask will override `KeyboardProperties.maskInputCharacters` in `SetGlobalProperties`. All other keyboard-related items in `SetGlobalProperties` will work normally.
+
+Also we would add a new private `SDLFormManager` sub-manager to the screen manager. This manager will handle Form-related capabilities, text, callbacks, etc. The `SDLScreenManager` itself will then provide a simple public API for presenting a Form to the screen.
+Here's the Objective-C version. This would be translated to Java and JavaScript as well
+```objc
+@interface SDLDisplayForm : SDLScreenManager
+/// Maps to `DisplayForm.initialText`. The text in the title bar of the form.
+@property (strong, nonatomic, nonnull) NSString *title;
+
+/// Maps to `DisplayForm.softButtons`. App defined SoftButtons for cancel, edit, submit, etc.
+@property (strong, nonatomic, nonnull) SoftButtonObjects *softButtons;
+
+/// Maps to `DisplayForm.fields`.
+@property (strong, nonatomic, nullable) Field *fields 
+
+/// Maps to `DisplayForm.errorMessage` If there's a problem with the input fields, this can be used to display information.
+@property (strong, nonatomic, nullable) NSString errorMessage
+
+- (instancetype)initWithPrimaryText:(nonnull NSString *)title fields:(nonnull Field *)fields softButtons:(nullable SoftButton *)softButtons errorMessage:(nullable NSString *)errorMessage;
+
+/// Cancels the RPC. If the view has not yet been sent to Core, it will not be sent. If the view is already presented on Core, the view will be immediately dismissed. Canceling an already presented view will only work if connected to Core versions 6.0+. On older versions of Core, the view will not be dismissed.
+
+- (void)cancel;
+@end
+```
```

Comment:
The requested revision was to specify that this proposal doesn't include manager changes. Please remove from this PR. These can be suggested as additional revisions once the proposal is back in review, if desired.